pkg/resource/configurator.go

```diff
@@ -56,7 +54,7 @@ func NewObjectMetaConfigurator(t runtime.ObjectTyper) *ObjectMetaConfigurator {
 // Configure the supplied Managed resource's object metadata.
 func (c *ObjectMetaConfigurator) Configure(_ context.Context, cm Claim, cs NonPortableClass, mg Managed) error {
```

Comment:
Should this configurator set the managed resource's external name to the value of the claim's `ExternalNameAnnotationKey`, if it exists at dynamic provisioning time?

Reply:
Another PR is coming up for use of `ExternalNameAnnotationKey`. Actually, I'll go ahead and remove that constant from this PR.

But the idea is that if get an annotation on the claim, we propagate it down to managed resource's annotation, then managed reconciler and/or external client will make use of that annotation. So, the name of the managed resource is not affected by this annotation on the claim.

Reply:
Ah I see. I agree that makes more sense than setting the actual name of the managed resource, given the annotation is about external names.